### PR TITLE
Clean up IME handling code

### PR DESCRIPTION
**Pull request by PoignardAzur**
Should fix the way repaints were requested every paint on some platforms.

This PR unfortunately comes with a lack of tests. TestHarness currently doesn't have good support for testing signals, but even if it did, this is an inherently visual feature that we don't really have a way to test visually.

I don't know how to trigger IME on my machine. (I'm using PopOS, a Debian variant.) If someone wants to either test this or help me get to speed, I'd appreciate it.

### CODE REVIEW

**Comment by PoignardAzur** [comment]
I just realized Textbox never actually calls register_as_text_input. Oops.

This is yet one more thing I'd like to test, but expressing it in a test is kinda awkward.

**Comment by DJMcNab** [comment]
I'll review when that is fixed.

**Comment by PoignardAzur** [comment]
Should be fixed, but I haven't tested it locally.

File: masonry/src/widget/widget_state.rs [hunk 1]
```diff
@@ -63,9 +62,16 @@ pub struct WidgetState {
     /// the baseline. Widgets that contain text or controls that expect to be
     /// laid out alongside text can set this as appropriate.
     pub(crate) baseline_offset: f64,
-    // TODO - Document
+    // TODO - Remove
     pub(crate) is_portal: bool,
 
+    /// Tracks whether widget is eligible for IME events.
+    /// Should be immutable after `WidgetAdded` event.
```
Comment (DJMcNab) [inline masonry/src/widget/widget_state.rs:69]: Why? I mean I realise that there is no way to unset this, but are there scenarios where we'd want this to change?

**Review by DJMcNab — APPROVED** [review]
I'm seeing this work. One thing I am running into is that `to_do_list` is extremely laggy, but that's still the case before this PR.

File: masonry/src/widget/widget_state.rs (hunk 1 above)
Comment (PoignardAzur) [reply masonry/src/widget/widget_state.rs:69]: Basically I'd like to be able to write our logic with the assumption that things like "is this focusable", "is this text-editable", etc, are immutable properties of a widget and don't need to be re-checked after creation.

**Pull request merged**